### ISSUE TITLE
Autodiff failed on latest ti.polar_decompose

### ISSUE THREAD

**Issue by erizmr** [taichi-dev/taichi#5664]
**Describe the bug**
The autodiff failed on latest `ti.polar_decompose` (v1.0.3).

**To Reproduce**

```py
import taichi as ti

ti.init(print_ir=True)
dim = 2
F_1 = ti.Matrix.field(dim,
                      dim,
                      dtype=ti.f32,
                      shape=1,
                      needs_grad=True)
F = ti.Matrix.field(dim,
                    dim,
                    dtype=ti.f32,
                    shape=1,
                    needs_grad=True)
loss = ti.field(dtype=ti.f32, shape=(), needs_grad=True)


@ti.kernel
def polar():
    r, s = ti.polar_decompose(F[0])
    F_1[0] += r


with ti.ad.Tape(loss=loss):
    polar()
```

**Log/Screenshots**

```
[ir.h:as@220] Assertion failure: is<T>()
```

**Comment by ailzhang** [comment]
Is this flaky? cc: @lin-hitonami

**Comment by lin-hitonami** [comment]
It fails stably, so it is not flaky. I'm looking into it.

**Comment by lin-hitonami** [comment]
I'm not very familiar with the autodiff code, but it seems that it's caused by the `stack` of some `AdStackPopStmt` is `LocalLoadStmt` instead of the expected `AdStackAllocaStmt`. @erizmr 
```
    $160 : if $159 {
      <f32> $161 = local load [ [$5[0]]]
      <f32> $162 = stack load top adj $161
      <f32> $163 : stack pop $161
      <f32> $164 = local load [ [$6[0]]]
      <f32> $165 = stack load top adj $164
      <f32> $166 : stack pop $164
      <f32> $167 = local load [ [$7[0]]]
      <f32> $168 = stack load top adj $167
      <f32> $169 : stack pop $167
      <f32> $170 = local load [ [$4[0]]]
      <f32> $171 = stack load top adj $170
      <f32> $172 : stack pop $170
      <f32> $173 = neg $162
      <f32> $174 = add $165 $168
      <f32> $175 : local store [$12 <- $174]
      <f32> $176 : local store [$13 <- $174]
      <f32> $177 = add $173 $171
      <f32> $178 : local store [$14 <- $177]
      <f32> $179 = neg $171
      <f32> $180 = add $162 $179
      <f32> $181 : local store [$15 <- $180]
    }
```

**Comment by erizmr** [comment]
Thanks for checking. @PhrygianGates is looking at this issue.

**Comment by ailzhang** [comment]
Per offline discussion, this failure also affects examples in difftaichi so adding to v1.1.0 release milestone. :D

**Comment by ailzhang** [comment]
fixes via #5692 , thanks @erizmr !
